params/state_upgrade.go

```diff
@@ -0,0 +1,44 @@
+// (c) 2023 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package params
+
+import (
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+type StateUpgrade struct {
+	blockTimestamp *big.Int
+
+	// Adds the specified amount to the balance of the specified address
+	AddToBalance map[common.Address]*math.HexOrDecimal256 `json:"addToBalance,omitempty"`
+
+	// Sets the specified storage slots of the specified addresses
+	// to the given values. Note that the value of common.Hash{} will
+	// remove the storage key
+	SetStorage map[common.Address]map[common.Hash]common.Hash `json:"setStorage,omitempty"`
+
+	// Sets the code of the specified contract to the given value
+	SetCode map[common.Address][]byte `json:"setCode,omitempty"`
+
+	// Deploys contracts with the specified creation bytecode to the
+	// specified addresses, instead of the normal rules for deriving
+	// the address of a created contract.
+	DeployContractTo []ContractDeploy `json:"deployContractTo,omitempty"`
```

Comment:
should we use same notation for `alloc` in genesis? https://github.com/ava-labs/subnet-evm/blob/master/core/genesis.go#L102 
like:
```
type StateUpgradeAccount struct {
	Code       []byte                      `json:"code,omitempty"`
	Storage    map[common.Hash]common.Hash `json:"storage,omitempty"`
	Balance    *big.Int                    `json:"balance" gencodec:"required"`
	Nonce      uint64                      `json:"nonce,omitempty"` 
}
```

and 
``` 
type StateUpgrade struct {
  BlockTimestamp *big.Int `json:"blockTimestamp"`
  StateUpgradeAlloc map[common.Address]GenesisAccount `json:"alloc"`
}
```

Reply:
This is also possible, I'd say this would be more inline with setting the balance to the specified amount instead of modifying the balance by the specified amount. Thoughts, @aaronbuchwald ?

Reply:
Changed it to something along those lines.
- Not sure we want to expose modifying nonce
- Kept BalanceChange semantics instead of setting balance.